x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -158,6 +160,12 @@ export class AbstractLayer implements ILayer {
 
     if (clonedDescriptor.joins) {
       clonedDescriptor.joins.forEach((joinDescriptor: JoinDescriptor) => {
+        if (joinDescriptor.right && joinDescriptor.right.type === SOURCE_TYPES.TABLE_SOURCE) {
```

Comment:
Why are you not cloning the join descriptor for a table source? I would think a cloned layer should carry over all information.

Reply:
discussed offline: throwing. This is for programmatic use only, no need to support cloning behavior from the UX.